packages/workspace/src/utils/rules/deleteFile.ts

```diff
@@ -2,6 +2,7 @@ import { forEach, FileEntry, Rule } from '@angular-devkit/schematics';
 
 /**
  * Remove a file from the Virtual Schematic Tree
+ * @deprecated This function will run through your whole workspace to find the file to delete. Please use `host.delete` instead for better performance
```

Comment:
@FrozenPandaz  I couldn't think of a better way to change this implementation without changing the signature or introducing more problems. So I think the deprecation note should be enough.